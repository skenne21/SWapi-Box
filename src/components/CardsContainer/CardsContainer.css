.CardsContainer {
  display: inline-block;
  height: 100vh;
  width: 80%;
}