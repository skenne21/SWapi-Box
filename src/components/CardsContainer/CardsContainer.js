import React from 'react';
import PropType from 'prop-types'
import Card from '../Card/Card.js';
import './CardsContainer.js';

const CardsContainer = ({cards, toggleFavorites, showFavorites, favorites, targetFavorites}) => {
  const createCards = cards.map((card, index) => 
    <Card 
    {...card} 
    key={card.name + index} 
    toggleFavorites={toggleFavorites}
    card={card}
    />)

  

  return (
    <div className='CardsContainer'> 
      {createCards}
    </div>
  )
}

CardsContainer.propTypes = {
  cards: PropType.array.isRequired, 
  toggleFavorites: PropType.func.isRequired
}

export default CardsContainer;