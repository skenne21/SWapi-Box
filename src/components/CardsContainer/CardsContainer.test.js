import React from 'react';
import {shallow} from 'enzyme';
import CardsContainer from './CardsContainer.js';

describe('CardsContainer', () => {
  const cards = [
    {
      name: "Luke Skywalker", 
      species: "Human", 
      homeWorldName: "Tatooine", 
      population: "200000"
    }, {
      name: "C-3PO",
      species: "Droid",
      homeWorldName: "Tatooine",
      population: "200000"
    }];

  it('should  not make any cards if array is empty', () => {
    let wrapper = shallow(<CardsContainer cards={[]}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should create 2 cards', () => {
    let wrapper = shallow(<CardsContainer cards={cards}/>);
    expect(wrapper).toMatchSnapshot();
  });
});

