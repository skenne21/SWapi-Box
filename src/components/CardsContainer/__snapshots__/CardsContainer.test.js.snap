// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsContainer should  not make any cards if array is empty 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardsContainer
    cards={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "CardsContainer",
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "CardsContainer",
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`CardsContainer should create 2 cards 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardsContainer
    cards={
        Array [
            Object {
              "homeWorldName": "Tatooine",
              "name": "Luke Skywalker",
              "population": "200000",
              "species": "Human",
            },
            Object {
              "homeWorldName": "Tatooine",
              "name": "C-3PO",
              "population": "200000",
              "species": "Droid",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          homeWorldName="Tatooine"
          name="Luke Skywalker"
          population="200000"
          species="Human"
/>,
        <Card
          homeWorldName="Tatooine"
          name="C-3PO"
          population="200000"
          species="Droid"
/>,
      ],
      "className": "CardsContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Luke Skywalker",
        "nodeType": "function",
        "props": Object {
          "homeWorldName": "Tatooine",
          "name": "Luke Skywalker",
          "population": "200000",
          "species": "Human",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "C-3PO",
        "nodeType": "function",
        "props": Object {
          "homeWorldName": "Tatooine",
          "name": "C-3PO",
          "population": "200000",
          "species": "Droid",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            homeWorldName="Tatooine"
            name="Luke Skywalker"
            population="200000"
            species="Human"
/>,
          <Card
            homeWorldName="Tatooine"
            name="C-3PO"
            population="200000"
            species="Droid"
/>,
        ],
        "className": "CardsContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Luke Skywalker",
          "nodeType": "function",
          "props": Object {
            "homeWorldName": "Tatooine",
            "name": "Luke Skywalker",
            "population": "200000",
            "species": "Human",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "C-3PO",
          "nodeType": "function",
          "props": Object {
            "homeWorldName": "Tatooine",
            "name": "C-3PO",
            "population": "200000",
            "species": "Droid",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
